nixos/modules/services/matrix/matrix-sliding-sync.nix

```diff
@@ -80,8 +80,11 @@ in
       } ];
     };
 
-    systemd.services.matrix-sliding-sync = {
-      after = lib.optional cfg.createDatabase "postgresql.service";
+    systemd.services.matrix-sliding-sync = rec {
+      after =
+        lib.optional cfg.createDatabase "postgresql.service"
+        ++ lib.optional config.services.matrix-synapse.enable "matrix-synapse.service";
```

Comment:
Does it fail because synapse itself is unreachable or because of $worker not being there? If the latter is the case, we'd probably want to use `matrix-synapse.target` here.

Reply:
I'm not actually sure which exact service is required. I've chosen to depend on `matrix-synapse.service` because that's always present unlike `matrix-synapse.target` which only exists when using workers and this way the configuration becomes a bit simpler. This is also how other services declare their dependency on Synapse (see [mautrix-whatsapp](https://github.com/NixOS/nixpkgs/blob/fa804edfb7869c9fb230e174182a8a1a7e512c40/nixos/modules/services/matrix/mautrix-whatsapp.nix#L103) or [appservice-discord](https://github.com/NixOS/nixpkgs/blob/fa804edfb7869c9fb230e174182a8a1a7e512c40/nixos/modules/services/matrix/appservice-discord.nix#L103) for examples).

Reply:
>  (see mautrix-whatsapp or appservice-discord for examples)

One of the reasons is probably that these modules are older than worker support.

> which only exists when using workers and this way the configuration becomes a bit simpler

What about checking if we have workers on the system and use either target or service?

Unless the sliding sync proxy strictly depends on the main process only no matter how/what workers are configured of course.

Reply:
I've added a new option `config.services.matrix-synapse.serviceUnit` now which is either `matrix-synapse.service` or `matrix-synapse.target`. That way, other modules don't need to check for worker support and can just add an `After`/`Wants` dependency on the new option. Do you think this is a good idea?

Reply:
Yes, I like that :+1: